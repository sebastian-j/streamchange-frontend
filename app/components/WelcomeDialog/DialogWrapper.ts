import styled, { keyframes } from 'styled-components';

const SlideDown = keyframes`
  0% {
    transform: translateY(-650px) translateX(-100px);
  }
  20% {
    transform: translateY(-650px) translateX(-100px);
  }
  100% {
    transform: translateY(0px) translateX(0px);
  }
`;

const DialogWrapper = styled.div`
  display: flex;
  flex-direction: column;
  position: absolute;
  top: 55%;
  left: 30%;
  transform: translate(-50%, -50%);

  @media (orientation: portrait) {
    left: 50%;
    top: 20%;
    width: 98%;
    transform: translate(-50%, 0%);
  }

  .dialog {
    animation: ${SlideDown} 0.7s ease-out;
    background-color: white;
    box-shadow: 0 11px 15px -7px rgba(0, 0, 0, 0.2),
      0 24px 38px 3px rgba(0, 0, 0, 0.14), 0 9px 46px 8px rgba(0, 0, 0, 0.52);
    border-radius: 4px;
    max-width: 600px;
    z-index: -1;

    .title {
      font-size: 1.25rem;
      font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
      font-weight: 500;
      line-height: 1.6;
      letter-spacing: 0.0075em;
      padding: 16px 24px;
    }
    .content {
      padding: 8px 24px;
      .text {
        color: rgba(0, 0, 0, 0.54);
        font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
        line-height: 1.5;
        margin-bottom: 12px;
      }
    }
    .actions {
      flex: 0 0 auto;
      display: flex;
      padding: 8px;
      align-items: center;
      justify-content: flex-end;
    }
  }
`;

export default DialogWrapper;
