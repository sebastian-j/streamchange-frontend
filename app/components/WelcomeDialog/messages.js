/*
 * WelcomeDialog Messages
 *
 * This contains all the text for the WelcomeDialog component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'streamchange.components.WelcomeDialog';

export default defineMessages({
  dialogTitle: {
    id: `${scope}.title`,
    defaultMessage: 'Choose stream',
  },
  videoInputLabel: {
    id: `${scope}.videoInput.label`,
    defaultMessage: 'Paste stream url',
  },
  example: {
    id: `${scope}.example`,
    defaultMessage: 'Example: https://www.youtube.com/watch?v=CBUBY45me_A',
  },
  saveBtn: {
    id: `${scope}.saveBtn`,
    defaultMessage: 'OK',
  },
  compatibilityInfo: {
    id: `${scope}.compatibilityInfo`,
    defaultMessage: 'App works only in Google Chrome',
  },
  hintTitle: {
    id: `${scope}.WelcomeHint.title`,
    defaultMessage: 'Did you know...',
  },
  invalidUrlError: {
    id: `${scope}.error.invalidUrl`,
    defaultMessage: 'This is not a URL to live stream or a YouTube video.',
  },
  notVideoError: {
    id: `${scope}.error.notVideo`,
    defaultMessage: 'There is no such live stream. Invalid url.',
  },
  notStreamError: {
    id: `${scope}.error.notStream`,
    defaultMessage: 'This is url for video. Paste live stream url.',
  },
  quotaExceededError: {
    id: `${scope}.error.quotaExceeded`,
    defaultMessage: 'The quota limit has been exceeded.',
  },
});
