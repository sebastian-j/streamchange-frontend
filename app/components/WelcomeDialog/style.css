.vertical {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    width: 100%;
}

.blur {
    background-color: rgba(0, 0, 0, 0.5);
    position: fixed;
    height: 100vh;
    width: 100vw;
}

.welcome-wrapper {
    display: flex;
    flex-direction: column;
    position: absolute;
    top: 55%;
    left: 50%;
    transform: translate(-50%, -50%);
}

.welcomeDialog {
    background-color: white;
    box-shadow: 0 11px 15px -7px rgba(0, 0, 0, 0.2), 0 24px 38px 3px rgba(0, 0, 0, 0.14), 0 9px 46px 8px rgba(0, 0, 0, 0.12);
    border-radius: 4px;
    max-width: 600px;
}

.welcomeDialog-title {
    font-size: 1.25rem;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-weight: 500;
    line-height: 1.6;
    letter-spacing: 0.0075em;
    padding: 16px 24px;
}

.welcomeDialog-content {
    padding: 8px 24px;
}

.welcomeDialog-actions {
    flex: 0 0 auto;
    display: flex;
    padding: 8px;
    align-items: center;
    justify-content: flex-end;
}

.welcomeDialog-textSecondary {
    color: rgba(0, 0, 0, 0.54);
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    line-height: 1.5;
    margin-bottom: 12px;
}

.welcomeHint {
    background-color: white;
    display: flex;
    flex-direction: row;
    box-shadow: 0 11px 15px -7px rgba(0, 0, 0, 0.2), 0 24px 38px 3px rgba(0, 0, 0, 0.14), 0 9px 46px 8px rgba(0, 0, 0, 0.12);
    border-radius: 4px;
    margin-top: 20px;
    max-width: 600px;
}

.welcomeHint-title {
    font-size: 1.2rem;
    font-weight: bold;
    display: block;
    margin: 10px;
}

.welcomeHint-content {
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    display: block;
    margin: 10px;
}

.compatibility-info {
    font-size: 3vw;
    text-align: center;
}