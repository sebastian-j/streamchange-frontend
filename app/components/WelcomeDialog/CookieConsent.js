import React, { useState } from 'react';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import CookieImage from './cookies.webp';

const Wrapper = styled.div`
  align-items: center;
  background: white;
  border-radius: 10px;
  box-shadow: 0px 10px 40px rgba(0, 0, 0, 0.32);
  bottom: 30px;
  display: flex;
  flex-direction: row;
  right: 30px;
  height: 10vh;
  position: fixed;
  transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  @media (orientation: portrait) {
    height: auto;
    right: 2%;
    width: 96%;
  }
`;

const Image = styled.img`
  margin: 1% 10px 1% 10px;
  height: 90%;
  @media (orientation: portrait) {
    height: 5vh;
  }
`;

const Description = styled.span`
  line-height: 1.1;
  max-width: 30vw;
  @media (orientation: portrait) {
    max-width: 58vw;
  }
`;
const AcceptButton = styled.button`
  background: white;
  border: 2px solid #0084b5;
  border-radius: 4px;
  color: #0084b5;
  cursor: pointer;
  font-weight: bold;
  font-size: 1.1em;
  padding: 0.3em 0.4em;
  margin: 15px;
  :hover {
    background-color: #0084b5;
    color: white;
  }
  @media (orientation: portrait) {
    margin: 5px;
  }
`;

const CookieConsent = () => {
  const [accepted, setAccepted] = useState(
    localStorage.getItem('cookieConsent') === 'true',
  );
  const handleAcceptClick = () => {
    localStorage.setItem('cookieConsent', 'true');
    setAccepted(true);
  };
  if (accepted) {
    return <div />;
  }
  return (
    <Wrapper>
      <Image src={CookieImage} alt="cookie" />
      <Description>
        <FormattedMessage {...messages.cookieInfo} />
      </Description>
      <AcceptButton onClick={handleAcceptClick} type="button">
        <FormattedMessage {...messages.cookieAcceptButton} />
      </AcceptButton>
    </Wrapper>
  );
};

export default CookieConsent;
