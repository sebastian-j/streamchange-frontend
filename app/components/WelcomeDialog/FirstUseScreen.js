import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';
import { createSelector } from 'reselect';
import { connect } from 'react-redux';
import Button from '@material-ui/core/Button';

import { makeSelectLocale } from '../../containers/LanguageProvider/selectors';
import { changeLocale } from '../../containers/LanguageProvider/actions';
import FlagPL from './flag-pl.png';
import FlagUK from './flag-uk.png';
import LightModeImg from './light-mode.png';
import DarkModeImg from './dark-mode.png';
import messages from './messages';

const Backdrop = styled.div`
  align-items: center;
  background-color: ${(props) => props.theme.bodyBackground};
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: fixed;
  height: 100vh;
  width: 100vw;
`;
const Title = styled.span`
  color: ${(props) => props.theme.staticTextColor};
  display: block;
  font-size: 2rem;
  margin-bottom: 1vh;
  margin-top: 5vh;
  text-align: center;
`;
const RadioInput = styled.input`
  display: none;
  &:checked + .box {
    background-color: #0068ad;
  }
`;

const Box = styled.div`
  width: 11vw;
  height: 11vw;
  background-color: #ffffffd9;
  transition: all 300ms ease;
  will-change: transition;
  display: inline-block;
  text-align: center;
  cursor: pointer;
  position: relative;
  font-family: sans-serif;
  ${({ left }) =>
    left &&
    `
    border-radius: 5px 0 0 5px;
  `}
  ${({ right }) =>
    right &&
    `
    border-radius: 0 5px 5px 0;
  `}
`;

const LangBox = styled.div`
  width: 11vw;
  height: 4vw;
  background-color: #ffffffd9;
  transition: all 250ms ease;
  will-change: transition;
  display: inline-block;
  text-align: center;
  cursor: pointer;
  position: relative;
  font-family: sans-serif;
  ${({ left }) =>
    left &&
    `
    border-radius: 4px 0 0 4px;
  `}
  ${({ right }) =>
    right &&
    `
    border-radius: 0 4px 4px 0;
  `}
`;

const Tile = styled.span`
  position: absolute;
  transform: translate(0, -50%);
  left: 5px;
  right: 5px;
  top: 85%;
  transition: all 300ms ease;
  font-size: 1.2vw;
  user-select: none;
`;

const LangTile = styled.span`
  position: absolute;
  transform: translate(0, -50%);
  left: 5px;
  right: 5px;
  top: 50%;
  transition: all 300ms ease;
  font-size: 1.2vw;
  user-select: none;
`;

const Flag = styled.img`
  margin-right: 10%;
  width: 20%;
`;

const ThemeImg = styled.img`
  margin-top: 10%;
  width: 96%;
`;
const StyledButton = styled(Button)`
  span {
    color: ${(props) => props.theme.materialButtonColor};
    font-size: 1.25rem;
  }
`;

const FirstUseScreen = (props) => {
  const [darkMode, setDarkMode] = useState(
    localStorage.getItem('darkMode') === 'true',
  );
  const [language, setLanguage] = useState(props.locale);

  const changeLanguage = (event) => {
    setLanguage(event.target.value);
    props.onLocaleToggle(event.target.value);
  };

  const changeTheme = (event) => {
    setDarkMode(event.target.value === '1');
    localStorage.setItem('darkMode', event.target.value === '1');
  };

  const save = () => {
    localStorage.setItem('locale', language);
    window.location.reload();
  };

  return (
    <Backdrop>
      <Title>
        <FormattedMessage {...messages.firstUseTitle} />
      </Title>
      <div className="tiles-grid">
        <Title>
          <FormattedMessage {...messages.firstUseSelectTheme} />
        </Title>
        <label htmlFor="themeLight">
          <RadioInput
            id="themeLight"
            type="radio"
            name="theme"
            value="0"
            onClick={changeTheme}
            checked={!darkMode}
          />
          <Box className="box" left>
            <ThemeImg src={LightModeImg} alt="light" />
            <Tile>
              <FormattedMessage {...messages.lightTheme} />
            </Tile>
          </Box>
        </label>
        <label htmlFor="themeDark">
          <RadioInput
            id="themeDark"
            type="radio"
            name="theme"
            value="1"
            onClick={changeTheme}
            checked={darkMode}
          />
          <Box className="box" right>
            <ThemeImg src={DarkModeImg} alt="dark" />
            <Tile>
              <FormattedMessage {...messages.darkTheme} />
            </Tile>
          </Box>
        </label>
      </div>
      <div className="tiles-grid">
        <Title>
          <FormattedMessage {...messages.firstUseSelectLang} />
        </Title>
        <label htmlFor="pl">
          <RadioInput
            id="pl"
            type="radio"
            name="language"
            value="pl"
            onClick={changeLanguage}
            checked={language === 'pl'}
          />
          <LangBox className="box" left>
            <LangTile>
              <Flag src={FlagPL} alt="pl" />
              <FormattedMessage {...messages.pl} />
            </LangTile>
          </LangBox>
        </label>
        <label htmlFor="en">
          <RadioInput
            id="en"
            type="radio"
            name="language"
            value="en"
            onClick={changeLanguage}
            checked={language === 'en'}
          />
          <LangBox className="box" right>
            <LangTile>
              <Flag src={FlagUK} alt="en" />
              <FormattedMessage {...messages.en} />
            </LangTile>
          </LangBox>
        </label>
      </div>
      <div>
        <StyledButton onClick={save} type="button">
          <FormattedMessage {...messages.firstUseSaveBtn} />
        </StyledButton>
      </div>
    </Backdrop>
  );
};

FirstUseScreen.propTypes = {
  onLocaleToggle: PropTypes.func,
  locale: PropTypes.string,
};

const mapStateToProps = createSelector(makeSelectLocale(), (locale) => ({
  locale,
}));

export function mapDispatchToProps(dispatch) {
  return {
    onLocaleToggle: (value) => dispatch(changeLocale(value)),
    dispatch,
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(FirstUseScreen);
