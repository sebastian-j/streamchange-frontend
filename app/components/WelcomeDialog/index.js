import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import { HINTS } from '../../config';
import './style.css';

const WelcomeDialog = props => {
  const [videoLink, setVideoLink] = useState('');
  const [isChrome, setIsChrome] = useState(true);
  const [hint, setHint] = useState('');

  const sendVideoLink = () => {
    if (typeof props.passVideo === 'function') {
      props.passVideo(videoLink);
    }
  };

  useEffect(() => {
    setIsChrome(!!window.chrome);
    setHint(HINTS[Math.floor(Math.random() * HINTS.length)]);
  }, []);

  const handleInputValueChange = event => {
    const { target } = event;
    const { value } = target;
    setVideoLink(value);
  };

  const handleKeyPress = e => {
    if (e.key === 'Enter') {
      sendVideoLink();
    }
  };

  if (isChrome) {
    return (
      <div className="blur">
        <div className="welcome-wrapper">
          <div className="welcomeDialog">
            <div className="welcomeDialog-title">
              Wybierz, na którym streamie organizujesz giveaway
            </div>
            <div className="welcomeDialog-content">
              <TextField
                autoFocus
                margin="dense"
                name="videoLink"
                onChange={handleInputValueChange}
                onKeyPress={handleKeyPress}
                id="videoLink"
                label="Wklej link do streama"
                type="text"
                value={videoLink}
                fullWidth
              />
              <div className="welcomeDialog-textSecondary">
                <span style={{ fontSize: '0.8rem' }}>
                  Przykład: https://www.youtube.com/watch?v=CBUBY45me_A
                </span>
                {props.error && (
                  <span style={{ display: 'block', color: '#bd0013' }}>
                    {props.error}
                  </span>
                )}
              </div>
            </div>
            <div className="welcomeDialog-actions">
              <Button onClick={sendVideoLink} color="primary">
                Dalej
              </Button>
            </div>
          </div>
          <div className="welcomeHint">
            <img
              src="/static/light-bulb.png"
              height="140px"
              alt="light bulb"
            />
            <div>
              <span className="welcomeHint-title">Czy wiesz, że...</span>
              <span className="welcomeHint-content">{hint}</span>
            </div>
          </div>
        </div>
      </div>
    );
  }
  return (
    <div className="vertical">
      <div className="compatibility-info">
        Aplikacja działa wyłącznie w Google Chrome
      </div>
    </div>
  );
};

WelcomeDialog.propTypes = {
  error: PropTypes.string,
  passVideo: PropTypes.func,
};

export default WelcomeDialog;
