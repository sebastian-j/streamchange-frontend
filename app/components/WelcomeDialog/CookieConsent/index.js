import React, { useState } from 'react';
import { FormattedMessage } from 'react-intl';

import { AcceptButton } from './components/AcceptButton';
import CookieImage from '../assets/cookies.webp';
import { Description } from './components/Description';
import { Image } from './components/Image';
import { Wrapper } from './components/Wrapper';
import messages from './messages';

const CookieConsent = () => {
  const [accepted, setAccepted] = useState(
    localStorage.getItem('cookieConsent') === 'true',
  );
  const handleAcceptClick = () => {
    localStorage.setItem('cookieConsent', 'true');
    setAccepted(true);
  };
  if (accepted) {
    return <div />;
  }
  return (
    <Wrapper>
      <Image src={CookieImage} alt="cookie" />
      <Description>
        <FormattedMessage {...messages.cookieInfo} />
      </Description>
      <AcceptButton onClick={handleAcceptClick} type="button">
        <FormattedMessage {...messages.cookieAcceptButton} />
      </AcceptButton>
    </Wrapper>
  );
};

export default CookieConsent;
