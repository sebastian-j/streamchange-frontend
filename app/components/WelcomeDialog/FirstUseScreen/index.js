import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { createSelector } from 'reselect';
import { connect } from 'react-redux';

import { makeSelectLocale } from '../../../containers/LanguageProvider/selectors';
import { changeLocale } from '../../../containers/LanguageProvider/actions';
import FlagPL from '../assets/flag-pl.png';
import FlagUK from '../assets/flag-uk.png';
import LightModeImg from '../assets/light-mode.png';
import DarkModeImg from '../assets/dark-mode.png';
import { Backdrop } from './components/Backdrop';
import { Box } from './components/Box';
import CookieConsent from '../CookieConsent';
import { LangBox } from './components/LangBox';
import { LangTile } from './components/LangTile';
import { RadioInput } from './components/RadioInput';
import { StyledButton } from './components/StyledButton';
import { Tile } from './components/Tile';
import { Title } from './components/Title';
import messages from './messages';

const FirstUseScreen = (props) => {
  const [darkMode, setDarkMode] = useState(
    localStorage.getItem('darkMode') === 'true',
  );
  const [language, setLanguage] = useState(props.locale);

  const changeLanguage = (event) => {
    setLanguage(event.target.value);
    props.onLocaleToggle(event.target.value);
  };

  const changeTheme = (event) => {
    setDarkMode(event.target.value === '1');
    localStorage.setItem('darkMode', (event.target.value === '1').toString());
  };

  const save = () => {
    localStorage.setItem('locale', language);
    window.location.reload();
  };

  return (
    <Backdrop>
      <Title>
        <FormattedMessage {...messages.firstUseTitle} />
      </Title>
      <div>
        <Title>
          <FormattedMessage {...messages.firstUseSelectTheme} />
        </Title>
        <label htmlFor="themeLight">
          <RadioInput
            id="themeLight"
            type="radio"
            name="theme"
            value="0"
            onChange={() => {}}
            onClick={changeTheme}
            checked={!darkMode}
          />
          <Box className="box left">
            <img src={LightModeImg} alt="light" />
            <Tile>
              <FormattedMessage {...messages.lightTheme} />
            </Tile>
          </Box>
        </label>
        <label htmlFor="themeDark">
          <RadioInput
            id="themeDark"
            type="radio"
            name="theme"
            value="1"
            onChange={() => {}}
            onClick={changeTheme}
            checked={darkMode}
          />
          <Box className="box right">
            <img src={DarkModeImg} alt="dark" />
            <Tile>
              <FormattedMessage {...messages.darkTheme} />
            </Tile>
          </Box>
        </label>
      </div>
      <div>
        <Title>
          <FormattedMessage {...messages.firstUseSelectLang} />
        </Title>
        <label htmlFor="pl">
          <RadioInput
            id="pl"
            type="radio"
            name="language"
            value="pl"
            onChange={() => {}}
            onClick={changeLanguage}
            checked={language === 'pl'}
          />
          <LangBox className="box left">
            <LangTile>
              <img src={FlagPL} alt="pl" />
              <FormattedMessage {...messages.pl} />
            </LangTile>
          </LangBox>
        </label>
        <label htmlFor="en">
          <RadioInput
            id="en"
            type="radio"
            name="language"
            value="en"
            onChange={() => {}}
            onClick={changeLanguage}
            checked={language === 'en'}
          />
          <LangBox className="box right">
            <LangTile>
              <img src={FlagUK} alt="en" />
              <FormattedMessage {...messages.en} />
            </LangTile>
          </LangBox>
        </label>
      </div>
      <div>
        <StyledButton onClick={save} type="button">
          <FormattedMessage {...messages.firstUseSaveBtn} />
        </StyledButton>
      </div>
      <CookieConsent />
    </Backdrop>
  );
};

FirstUseScreen.propTypes = {
  onLocaleToggle: PropTypes.func,
  locale: PropTypes.string,
};

const mapStateToProps = createSelector(makeSelectLocale(), (locale) => ({
  locale,
}));

export function mapDispatchToProps(dispatch) {
  return {
    onLocaleToggle: (value) => dispatch(changeLocale(value)),
    dispatch,
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(FirstUseScreen);
