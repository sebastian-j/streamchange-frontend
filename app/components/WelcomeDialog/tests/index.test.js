import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import WelcomeDialog from '../index';

const renderer = new ShallowRenderer();

describe('<WelcomeDialog />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<WelcomeDialog passVideo={() => 0} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
