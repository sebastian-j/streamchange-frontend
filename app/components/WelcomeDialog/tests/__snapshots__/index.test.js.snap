// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeDialog /> should render and match the snapshot 1`] = `
<div
  className="blur"
>
  <div
    className="welcome-wrapper"
  >
    <div
      className="welcomeDialog"
    >
      <div
        className="welcomeDialog-title"
      >
        Wybierz, na którym streamie organizujesz giveaway
      </div>
      <div
        className="welcomeDialog-content"
      >
        <ForwardRef(WithStyles)
          autoFocus={true}
          fullWidth={true}
          id="videoLink"
          label="Wklej link do streama"
          margin="dense"
          name="videoLink"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
        <div
          className="welcomeDialog-textSecondary"
        >
          <span
            style={
              Object {
                "fontSize": "0.8rem",
              }
            }
          >
            Przykład: https://www.youtube.com/watch?v=CBUBY45me_A
          </span>
        </div>
      </div>
      <div
        className="welcomeDialog-actions"
      >
        <ForwardRef(WithStyles)
          color="primary"
          onClick={[Function]}
        >
          Dalej
        </ForwardRef(WithStyles)>
      </div>
    </div>
    <div
      className="welcomeHint"
    >
      <img
        alt="light bulb"
        height="140px"
        src="https://streamchange.pl/static/light-bulb.png"
      />
      <div>
        <span
          className="welcomeHint-title"
        >
          Czy wiesz, że...
        </span>
        <span
          className="welcomeHint-content"
        >
          
        </span>
      </div>
    </div>
  </div>
</div>
`;
